.github/workflows/deploy-sphinx-docs-to-ghpages.yml

```diff
@@ -71,15 +75,71 @@ jobs:
       - name: Upload artifact
         uses: actions/upload-pages-artifact@v1
         with:
+          name: documentation
           path: ${{ inputs.path-doc }}/build/html
-      
+  
+  # TODO: this job should have a own reusable workflow
+  new-artifact:
+    needs: [build]
+    if: ${{ ( inputs.trigger-label == 'latest' ) || ( inputs.trigger-label == 'stable' ) }} 
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - run: ls # TODO: remove this debugging line 
+    - uses: actions/download-artifact@v3
+      with:
+        name: documentation  
+    - run: | # TODO: remove this debugging line
+        ls 
+        pwd 
+    - name: generate the ${{inputs.trigger-label}} documentation 
+      run: |
+        label=${{inputs.trigger-label}}
+        mkdir new_docs
+        mv artifact.tar "${label}.tar" 
+        mv "${label}.tar" new_docs/
+        cd new_docs
+        ls # TODO: remove debugging line 
+        mkdir $label
+        tar -xf "${label}.tar" -C "${label}/" 
+        ls ../new_docs # TODO: remove this debugging line
+    - name: generate the remaning documentation
+      run: | 
+        label=${{inputs.trigger-label}} 
+        cd new_docs
+        if [[ ${label} == "latest" ]]; 
+          then 
+            missing_doc="stable"
+          else
+            missing_doc="latest"
+        fi
+        URL="https://qibogang.github.io/qibo/${missing_doc}/${missing_doc}"
+        mkdir "$missing_doc"
+        # Check if it is possible to connect to the website 
+        # Otherwise it copies the existing docs
+        if wget --spider "${URL}" 2>/dev/null; then 
+            wget "$URL"  
+        else
+            cp "${label}.tar" "${missing_doc}.tar"
+        fi   
+        tar -xf "${missing_doc}.tar" -C "${missing_doc}/"
+        echo "generate remaning doc" # TODO: remove this debugging line
+        ls # TODO: remove this debugging line
+        echo "let see in stable folder" # TODO: remove this debugging line
+        ls stable # TODO: remove this debugging line
+    - name: Upload artifact
+      uses: actions/upload-pages-artifact@v1
```

Comment:
Here is correct, the final one you want to be an actual `upload-pages-artifact`

Reply:
If the action requests an `index.html` write a simple one that redirects to `/<name>/latest`
https://stackoverflow.com/a/5411567/8653979

You can also inline in the workflow for the time being, e.g. with a [here doc](https://en.wikipedia.org/wiki/Here_document). Later on we will find a better place to store the template, most likely in this repo.